File: Emulator/src/main/core/game/node/entity/lock/LockElapse.java
package core.game.node.entity.lock;

import core.game.node.Node;

/**
 * The interface Lock elapse.
 */
public interface LockElapse {

    /**
     * Elapse.
     *
     * @param node the node
     * @param lock the lock
     */
    public void elapse(Node node, Lock lock);

}

File: Emulator/src/main/core/game/node/entity/lock/LockElapse.kt
package core.game.node.entity.lock

import core.game.node.Node

/**
* Functional interface representing a callback that executes when a [Lock] expires.
*/
fun interface LockElapse {

    /**
     * Called when the associated [Lock] elapses.
     *
     * @param node The node associated with the lock.
     * @param lock The lock that has expired.
     */
    fun elapse(node: Node, lock: Lock)
}